python/pypto/runtime/device_runner.py

```
@@ -123,22 +128,51 @@ def ensure_pto_isa_root(commit: str | None = None, clone_protocol: str = "https"
+                f"Cloning pto-isa from {primary_url} timed out after "
+                f"{_PTO_ISA_PRIMARY_CLONE_TIMEOUT}s; falling back to {fallback_url}"
+            )
+        except Exception as e:
+            logger.warning(f"Cloning pto-isa from {primary_url} failed: {e}; falling back to {fallback_url}")
+
+        if primary_failed and not include_dir.exists():
+            # Clean up any partial clone before retrying with the fallback.
+            if clone_path.exists():
+                shutil.rmtree(clone_path, ignore_errors=True)
+            try:
+                result = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)